### ISSUE TITLE
Sync Broken - OneDrive

### ISSUE THREAD

**Issue by ZorkC64-128** [laurent22/joplin#1138]
### Operating system

- Windows
- Android

### Application

- Desktop
- Mobile

Seems like Onedrive is now failing to sync  between android and desktop clients or the entire notebook is becoming corrupted.  New notes are not synchronizing back and forth and images are missing from the desktop client.

I have however noticed this issue after I created 'folders' for the notebooks. I don't see folders in the android client.

![screenshot 2](https://user-images.githubusercontent.com/42756324/51431953-2cdf8500-1bfe-11e9-9548-5fc7e164e839.PNG)
![screenshot_2019-01-19-15-10-37](https://user-images.githubusercontent.com/42756324/51431954-2cdf8500-1bfe-11e9-91aa-a3d9886e22f7.png)
![screenshot_2019-01-19-15-17-32](https://user-images.githubusercontent.com/42756324/51431955-2d781b80-1bfe-11e9-91d8-23c9f427fd8c.png)

**Comment by laurent22** [comment]
Could you check the file .resource/a90542772...... on OneDrive - what size is it? Also is there a storage limit on OneDrive and are you below it?

**Comment by ZorkC64-128** [comment]
2.53 kilobytes on the a905file.  I have 23GB out of 30GB used on this 'OneDrive' account. I have almost complete notebook failure. 

Re-installed Joplin, cleared local directory and replaced the note files back. I think it may have fixed the sync problem

However, now I have a bunch of notes that are getting ignored. If I import them in they are there but note decrypted. 

![image](https://user-images.githubusercontent.com/42756324/51445541-54a11c80-1cd4-11e9-9baa-defca20bf5c5.png)

Hoping I can recover these notes.. I have had synchronization problems like this with 'OneDrive' in the past with other software.

**Comment by laurent22** [comment]
> If I import them in they are there but not decrypted.

The files on the sync target aren't supposed to be imported like that, they are only for syncing. Also in your case they are encrypted so you'll need to provide the master key password.

To do that, I would take all these files and put them on the new sync target directory (something like /Apps/Joplin on OneDrive), then sync. At that point it should ask you for the master key password and, once you provide it, it should decrypt all these notes.

As far as I can see this is not really a bug in Joplin so I'm closing this issue, but feel free to add more comments if needed.

**Comment by ZorkC64-128** [comment]
Doing this course of action ends up causing Onedrive to refuse synchronization completely. It first creates a 'Joplin 1' folder but regardless of how these Joplin directories are deleted or replaced,it looks for a folder called '.sync'. This folder can exist however cannot be seen by Joplin and therefore all these notes remain unrecoverable.

**Comment by ZorkC64-128** [comment]
Day 5 Report ;

I guess its misleading to say I think I've had this same kind of problems with OneDrive on other apps. My apologies for saying that. Those are separate issues and currently those apps are functioning fine Unfortunately, the problem is still existing right here on home turf and we have two separate issues going on.

1 - Synchronization with One Drive with this app has a tenancy to  fail.

2 - When this happens, recovery of notes are impossible via any current method described above.

I was able to recover about 30% of the notes in notebook.  Oddly some notes will show up while others continue to fail.  If you need any other information I'm here to help. I'm going to report any updates here.

Again my apologies, I know development is busy with other features and fixes but I did not personally cause these bugs myself, I simply trusted the software to protect sensitive data and now stuck.

**Comment by ZorkC64-128** [comment]
Synchronization with 'One drive' is entirely broken. I am continuously getting this error now.

![joplin-screenshot-4](https://user-images.githubusercontent.com/42756324/51640816-86172380-1f32-11e9-8172-7d6c5f684254.PNG)

Further investigation reveals that my missing notes were deleted by Joplin and are no longer recoverable. Apparently Joplin has the ability to delete notes automatically on its own in certain scenarios. The lack of user intervention of how Joplin goes about cleaning house is a very dangerous and messy way of synchronization. Why would should Joplin delete notes relevant to me?

I take data loss pretty seriously.  As a supporter I will try and replicate and pinpoint how Joplin is creating data loss though its synchronization. However given the situation and the outcome, at this point I cannot trust my data to be secure with-in Joplin. Again, anything you need let me know.

**Comment by laurent22** [comment]
I had weird bugs with OneDrive when I was moving notes around or renaming folders. It seems OneDrive keeps some internal representation that's different from what it displays, and then displays strange errors. The behaviour of its API, unlike Dropbox or WebDAV, is quite opaque in general.

From your error I see it's trying to access /Apps/Joplin1 (notice the "1") - does that folder actually exist? How about /Apps/Joplin, does it exist and does it contain something? It's hard to tell what OneDrive is doing because this /Apps/Joplin is assigned by OneDrive, not by Joplin so I don't know why it's appending a "1".

I think the best would be to export all your data to JEX, then delete the local data, delete all the Joplin* folders on OneDrive. Then re-import the JEX file and re-sync. That will allow starting over from a clean state. Also I think you can restore data in OneDrive, it keeps some history if I'm not mistaken.

**Comment by ZorkC64-128** [comment]
Thanks for your reply. I believe you are right in saying ' It seems 'OneDrive' keeps some internal representation that's different from what it displays, and then displays strange errors. ' Deleting all of Joplin on 'OneDrive' is not working. It complains of the same error time and time again. 

As far as whats left of the notebook, exporting to 'JEX' may help preserve what can be accessed. I will give that a try and post here the results. I'll look at 'OneDrive' and see if I can find some sort of restore option, again thanks. I did not think of all that. 

Also too, I witnessed Joplin deleting notes.  This is the scariest part of the whole problem. 

When the sync went wrong over 'OneDrive' I noticed notes getting deleted as each device synchronized. Even when the notes were still available on other devices, once synchronization began Joplin was deleting notes. It shows this operation in the status. Thus all that it was deleting was lost I would assume. 

 Thanks again for your support I'm gong to continue investigating this. I hope I can give some concrete data to help make Joplin better. I think maybe flagging these conflicts and storing them in a separate folder might be a solution over deleting the files.  Anything I can do let me know.
